### PR TITLE
Physics/interaction

### PR DESCRIPTION
**Pull request by SomeOne768**

### CODE REVIEW

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.cpp [hunk 1]
```diff
@@ -30,10 +30,10 @@ namespace sne
             return _points[i];
         }
 
-        bool BoxCollider::hasPoint(const glm::vec3 &p) const
+        bool BoxCollider::hasPoint(const glm::vec3 &p, float eps) const
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.cpp]: eps not used

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp [hunk 2]
```diff
@@ -7,7 +7,7 @@
  *
  */
 
-#include "architecture/Component.hpp"
+#include "Collider.hpp"
 #include <vector>
 #include <glm/glm.hpp>
 #include <glm/gtc/type_ptr.hpp>
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/BoxCollider.hpp:14]: include lib files before the one from this project

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp [hunk 3]
```diff
@@ -0,0 +1,152 @@
+#include "PhysicObject.hpp"
+#include "../architecture/util.hpp"
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp]: put the include files inside hpp file

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+#include "SphereCollider.hpp"
+#include "BoxCollider.hpp"
+#include "SAT.hpp"
+
+namespace sne
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp]: sne::saintNecPhysics

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                throw std::exception_ptr();
+
+            try
+            {
+                if (!_collider->collide(obj._collider))
+                    return;
+            }
+            catch (const SATIllegalUseException &e)
+            {
+                std::cout << e.what() << '\n';
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp]: IS it possible to add details for the exception -> get the gameobject name for example

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.hpp [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+ * @file PhysicObject.hpp
+ * @brief PhysicObject representation of the object in the game.
+ * @author Abdeljalil ZOGHLAMI
+ * @date november 17th 2023.
+ */
+
+#include "architecture/Component.hpp"
+#include "Collider.hpp"
+#include <glm/glm.hpp>
+#include <exception>
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.hpp:12]: put lib before

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.hpp [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+namespace sne
+{
+    const int FPS = 10;
+    namespace saintNecPhysics
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.hpp]: sne::saintNecPhysics

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SphereCollider.cpp [hunk 8]
```diff
@@ -1,4 +1,5 @@
 #include "SphereCollider.hpp"
+#include "SAT.hpp"
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/SphereCollider.cpp]: put include in hpp

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp (hunk 3 above)
Comment (SomeOne768) [reply Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/physics/PhysicObject.cpp]: Some are usefull only in the cpp (using of algo). Are you sure to put whole of them into the hpp ?

**Review by antonindevidal — APPROVED** [review]

File: Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/architecture/util.cpp [hunk 9]
```diff
@@ -67,7 +67,15 @@ namespace glm
 
 namespace sne
 {
-    double norm(const glm::vec3 &v)
+    bool areSimilar(const glm::vec3 &v1, const glm::vec3 &v2, float eps)
+    {
+        if ((v1 - eps) <= v2 && v2 <= (v1 + eps))
```
Comment (antonindevidal) [inline Saint-Nec-Engine/Saint-Nec-Engine-Lib/src/architecture/util.cpp]: can be refactored

**Review by antonindevidal — APPROVED** [review]

**Pull request merged**